web/blueprints/finance/__init__.py

```diff
@@ -123,18 +129,92 @@ def actions(activity_id):
 @nav.navigate(u"Bankkontobewegungen importieren")
 def bank_accounts_import():
     form = BankAccountActivitiesImportForm()
+    form.account.choices = [ (acc.id, acc.name) for acc in BankAccount.q.all()]
+    transactions = [] # new transactions which would be imported
+    old_transactions = [] # transactions which are already imported
     if form.validate_on_submit():
+        # login with fints
+        bankaccount = BankAccount.q.get(form.account.data)
         try:
-            csv_data = StringIO(form.csv_file.data.read().decode('utf-8'))
-            finance.import_bank_account_activities_csv(csv_data, form.expected_balance.data)
+            fints = FinTS3PinTanClient(
+                bankaccount.routing_number,
+                form.user.data,
+                form.pin.data,
+                bankaccount.fints_endpoint
+            )
+
+            acc = None
+            for account in fints.get_sepa_accounts():
+                if account.iban == bankaccount.iban:
+                    acc = account
+                    break
+            if acc is None:
+                raise KeyError('BankAccount with IBAN {} not found.'.format(
+                    bankaccount.iban)
+                )
+            if bankaccount.last_updated_at is not None:
+                start_date = bankaccount.last_updated_at.date()
+            else:
+                start_date = date(2018,1,1)
+            statement = fints.get_statement(acc, start_date, date.today())
+            flash(
+                "Transaktionen vom {} bis {}.".format(start_date, date.today()))
+        except FinTSDialogError:
+            flash(u"Ungültige FinTS-Logindaten.", 'error')
+            statement = []
+        except KeyError:
+            flash(u'Das gewünschte Konto kann mit diesem Online-Banking-Zugang\
+                    nicht erreicht werden.', 'error')
+            statement = []
+
+
+        for transaction in statement:
+            iban = transaction.data['applicant_iban'] if \
+                transaction.data['applicant_iban'] is not None else ''
+            bic = transaction.data['applicant_bin'] if \
+                transaction.data['applicant_bin'] is not None else ''
+            other_name = transaction.data['applicant_name'] if \
+                transaction.data['applicant_name'] is not None else ''
+            new_activity = BankAccountActivity(
+                bank_account_id=bankaccount.id,
+                amount=int(transaction.data['amount'].amount*100),
+                reference=transaction.data['purpose'],
+                original_reference=transaction.data['purpose'],
+                other_account_number=iban,
+                other_routing_number=bic,
+                other_name=other_name,
+                imported_at=datetime.now(),
```

Comment:
I think it would be better to use the same timestamp for the whole batch of imported transactions.

Apart from that I am not sure if `datetime.now()` is the right way to get the current time. With the changes proposed in #162 it certainly wouldn't, but as the changes are still to be implemented, and me being not aware of the current timezone situation, it would be great, if a more seasoned Pycroft developer (e.g. @sebschrader) could comment on this.

Reply:
(This comment is yet not outdated.)

Reply:
I agree that it would be better to use the same timestamp for the whole batch. But before changing this in code I'll wait for comments on the way of getting the current time.